### PR TITLE
Fix Pinecone Test

### PR DESCRIPTION
**Pull request by karansampath**
Fixes the `test_pinecone_scan` integration test that was not working due to eventual consistency issues

### CODE REVIEW

File: lib/sycamore/sycamore/tests/integration/connectors/pinecone/test_pinecone_scan.py [hunk 1]
```diff
@@ -54,3 +60,24 @@ def test_pinecone_scan():
...
+    """
+    ready = False
+    timeout = 30
+    deadline = time.time() + timeout
+    index = client.Index(index_name)
+    while not ready:
+        try:
+            desc = dict(index.fetch(ids=[doc_id], namespace=namespace)["vectors"]).items()
+            if len(desc) > 0:
+                ready = True
```
Comment (HenryL27) [inline lib/sycamore/sycamore/tests/integration/connectors/pinecone/test_pinecone_scan.py:77]: suggest:
`while time.time() < deadline:`
...
`if len(desc) > 0: return`

**Review by HenryL27 — APPROVED** [review]
lgtm. pay no mind to the suggestion unless you want to.
I guess let's see if it gets the test to pass

File: lib/sycamore/sycamore/tests/integration/connectors/pinecone/test_pinecone_scan.py [hunk 2]
```diff
@@ -66,18 +66,15 @@ def wait_for_write_completion(client: PineconeGRPC, index_name: str, namespace:
...
             desc = dict(index.fetch(ids=[doc_id], namespace=namespace)["vectors"]).items()
             if len(desc) > 0:
-                ready = True
+                return
         except PineconeException:
             # NotFoundException means the last document has not been entered yet.
             pass
         time.sleep(1)
-        if time.time() > deadline:
-            raise RuntimeError(f"Pinecone failed to write results in {timeout} seconds. Doc_id: {doc_id}")
```
Comment (HenryL27) [inline lib/sycamore/sycamore/tests/integration/connectors/pinecone/test_pinecone_scan.py]: still want to raise this after the loop I think? I guess you can just yolo the read, eh

Comment (karansampath) [reply lib/sycamore/sycamore/tests/integration/connectors/pinecone/test_pinecone_scan.py]: Yeah I think that'll be better, there might be multiple reasons for it failing

**Pull request merged**